### ISSUE TITLE
need wait member cluster informer cache sync after controller started

### ISSUE THREAD

**Issue by wu0407** [karmada-io/karmada#4065]
**What happened**:
Many errors on the controller manager start but do not affect the controller work.  I think it needs to be optimized.
```
execution_controller.go:121] Failed to sync work(xxxx) to cluster(xxx): the informer of cluster(xxxx) has not been initialized
E0914 03:52:37.850177       1 execution_controller.go:202] Failed to create or update resource(/xxxx) in the given member cluster sh5-online, err is the informer of cluster(xxxxx) has not been initialized
```

**What you expected to happen**:
Don't have any errors

**How to reproduce it (as minimally and precisely as possible)**:
Delete karmada-controller manager pod and see the logs. 

**Anything else we need to know?**:

**Environment**:
- Karmada version: 1.7.0
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

**Comment by chaunceyjiang** [comment]
This issue is known. See https://github.com/karmada-io/karmada/blob/4ee4b22f056c2768c96752dc0426c4291ae1e1b2/pkg/controllers/status/cluster_status_controller.go#L191-L196

**Comment by RainbowMango** [comment]
Yes, the execution_controller now needs to wait for cluster_status_controller to initialize the informer, I don't have a clue how to improve it yet :(

**Comment by wu0407** [comment]
The user uses an alert rule with `work_sync_workload_duration_seconds_count` may take care of this. It will cause a false alarm.

**Comment by wu0407** [comment]
Another noise  is "Operation cannot be fulfilled on "xxx": the object has been modified; please apply your changes to the latest version and try again"
